cores/esp8266/FS.h

```diff
@@ -67,13 +67,14 @@ class File : public Stream
     size_t readBytes(char *buffer, size_t length) override {
         return read((uint8_t*)buffer, length);
     }
-    size_t read(uint8_t* buf, size_t size);
+    int read(uint8_t* buf, size_t size) override;
     bool seek(uint32_t pos, SeekMode mode);
     bool seek(uint32_t pos) {
         return seek(pos, SeekSet);
     }
     size_t position() const;
     size_t size() const;
+    virtual ssize_t streamSize() override { return size() - position(); }
```

Comment:
While technically I can see it, what would a negative `streamSize()` mean, anyway???

Reply:
program error as position is beyond the size?

Reply:
If the question is about the choice of the return type (I had the choice in this case!),
I was thinking while writing it that a signed type is better, because, and this has been discussed elsewhere in this repo not a long time ago, we can read here and there things like `if (retValue == (size_t)-1)`. Using `ssize_t` can avoid that.

Regarding the value, `someFunctionReturningASize()<0` is usually used to indicate an error, and `==0` is EOF(FS), TryAgain(non-blocking network op) or closed/timeout(blocking network op).

Reply:
> into a large weird number

I'm not sure of that because `(ssize_t)((size_t)1 - (size_t)2) == -1`.
I added the cast anyway for less confusion.